security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -432,6 +434,15 @@ func (sc *SecretCache) generateSecret(ctx context.Context, token, resourceName s
 			return nil, fmt.Errorf("cannot find secret %s for ingress gateway", resourceName)
 		}
 
+		if strings.HasSuffix(resourceName, secretfetcher.IngressGatewaySdsCaSuffix) {
+			return &model.SecretItem{
+				ResourceName: resourceName,
```

Comment:
You need to document this in istio/api - that if mTLS is used and the credentialName is specified, with SDS enabled, then we expect the secret to be of the form credentialName-cacert

Reply:
Will update istio/api once this PR is in.